timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/LockCollection.java

```diff
@@ -16,12 +16,22 @@
 package com.palantir.atlasdb.timelock.lock;
 
 import com.palantir.lock.LockDescriptor;
+import java.util.Optional;
 import java.util.Set;
 
 final class LockCollection {
```

Comment:
I think I understand roughly what you're trying to do here: you want ALL the lock types to be kept together, so we can draw attention to the weird design where everything is a string and TECHNICALLY you could grab a lock on the descriptor of "ImmutableTimestamp:1234" through the ExclusiveLockCollection.

Is your thinking that this class will now basically understand the "patterns" that all locks can have in their descriptors, and basically reject an acquire request through ExclusiveLockCollection that COULD potentially conflict with what you should acquire through ImmutableTimestampTracker.

So this class effectively maintains the invariant that "it's impossible for a LockDescriptor to live in more than one collection, so if it's in ExclusiveLockCollection it will never be in ImmutableTimestampTracker".

Right?

If that's your aim, I wholeheartedly support that, but let's think of a name for LockCollection (because it's no longer serving us) and add a tiny bit of javadoc to explain. Lemme talk to ChatGPT to suggest some good names

Reply:
Alright after consulting ChatGPT, I feel like this now should be called "LockManager", because it coordinates across different types of locks, but also maintains a key safety principle that you can't aquire the same lock descriptor through 2 different methods.